src/lib.rs

```diff
@@ -458,21 +458,13 @@ where
         T: Translator<P, Q, E>;
 }
 
-/// Either a key or a keyhash
-pub enum ForEach<'a, Pk: MiniscriptKey> {
-    /// A key
-    Key(&'a Pk),
-    /// A keyhash
-    Hash(&'a Pk::Hash),
-}
+/// Either a key or keyhash, but both contain Pk
+pub struct ForEach<'a, Pk: MiniscriptKey>(&'a Pk);
```

Comment:
Note to reviewers. We deliberately do not iterate over RawPkH here. In a follow-up PR, we will make `Miniscript::from_ast` fail when we reach a `RawPkh` terminal. 

We can update this whenever partial descriptors are supported upstream

Reply:
I remember @sanket1729 and I discussed abouting removing this later on, but I don't see any reason why I can't do it here. I'm pushing another commit on top of 5cb2bcf to remove `ForEach`. If accepted, I can squash the commit anytime!